File: php/muse/data_structures/LinkedQueueTest.php
<?php

declare(strict_types=1);

require_once "muse/data_structures/LinkedQueue.php";
require_once "muse/util/TestRunner.php";

use muse\data_structures\LinkedQueue;
use muse\util\TestRunner;

function testLinkedQueue(): bool
{
    $size = 8192;

    $queue = new LinkedQueue();

    for ($i = 1; $i <= $size; $i++) {
        $queue->enqueue($i);
    }

    if ($queue->size() != $size) {
        return false;
    }

    for ($i = 1; $i <= $size; $i++) {
        if ($queue->peek() != $i) {
            return false;
        }
        $queue->dequeue();
    }

    return $queue->isEmpty();
}

if (count(debug_backtrace()) == 0) {
    TestRunner\pick("testLinkedQueue");
}

File: python/muse/data_structures/linked_queue_test.py
from muse.data_structures import tests
from muse.data_structures.linked_queue import LinkedQueue
from muse.util import test_runner


def test_linked_queue() -> bool:
    return tests.queue_derive(LinkedQueue)()


def main() -> None:
    test_runner.pick(test_linked_queue)


if __name__ == "__main__":
    main()
